controllers/userController.js

```diff
@@ -77,40 +163,55 @@ const controller = {
           populate: { path: 'project_id', select: '-__v -user_id -_id' },
         })
         .lean();
-      const contributedProjects = [];
-      for (let i = 0, len = contributedJobs.length; i < len; i++) {
-        contributedProjects[i] = contributedJobs[i]?.contributor_id?.project_id;
+      const userProjectsAccepted = [];
+      for (let i = 0, len = jobsAccepted?.length; i < len; i++) {
+        userProjectsAccepted[i] = jobsAccepted[i]?.contributor_id?.project_id;
       }
-      const appliedProjects = [];
-      for (let i = 0, len = contributedJobs.length; i < len; i++) {
-        appliedProjects[i] = appliedJobs[i]?.contributor_id?.project_id;
+
+      return res.json(userProjectsAccepted);
+    } catch (error) {
+      console.log(error);
+      return res.status(500).json({
+        error: 'Failed to fetch contributed projects of this user from database',
+      });
+    }
+  },
+  showUserProjectsFollowing: async (req, res) => {
+    const username = req.params.username;
+    try {
+      const profileOwner = await UserModel.findOne({ username }, { __v: 0, hash: 0 }).lean();
+      if (!profileOwner) {
+        return res.status(404).json();
       }
       const followingProjectsRelationship = await ProjectsRelationshipModel.find(
         { user_id: profileOwner._id, state: 'published' },
```

Comment:
there is no state under ProjectsRelationshipModel right?

Reply:
updated